### PR TITLE
[Merged by Bors] - feat(Mathlib/Tactic/Basic): add `triv` tactic

### PR DESCRIPTION
**Pull request by kbuzzard**
My first tactic, please check carefully!

mathlib3 tactic is https://github.com/leanprover-community/mathlib/blob/903e8941f0862f6c34f28babbb607842a7f8a6c7/src/tactic/interactive.lean#L663

### CODE REVIEW

**Comment by j-loreaux** [comment]
According to the mathlib3 documentation, `triv` differs from `trivial` in that it does not try to apply contradiction. So doesn't this need to avoid that scenario too?

File: Mathlib/Tactic/Basic.lean [hunk 1]
```diff
@@ -256,3 +256,11 @@ elab "fapply " e:term : tactic =>
...
 elab "eapply " e:term : tactic =>
   evalApplyLikeTactic (Meta.apply (cfg := {newGoals := ApplyNewGoals.nonDependentOnly})) e
+
+/--
+Tries to solve the goal using a canonical proof of `true` or the `reflexivity` tactic.
+-/
+syntax (name := triv) "triv" : tactic
+
+macro_rules
+  | `(tactic| triv) => `(tactic| first | exact trivial | rfl | fail "triv tactic failed")
```
Comment (digama0) [inline Mathlib/Tactic/Basic.lean:266]:
```suggestion
macro (name := triv) "triv" : tactic =>
  `(tactic| first | exact trivial | rfl | fail "triv tactic failed")
```

**Comment by semorrison** [comment]
@kbuzzard, it's easier for everyone if you make PRs direct from the leanprover-community repository. (I've just merged master for you.)

File: Mathlib/Tactic/Basic.lean [hunk 2]
```diff
@@ -260,6 +260,12 @@ elab "fapply " e:term : tactic =>
 elab "eapply " e:term : tactic =>
   evalApplyLikeTactic (Meta.apply (cfg := {newGoals := ApplyNewGoals.nonDependentOnly})) e
 
+/--
+Tries to solve the goal using a canonical proof of `true` or the `reflexivity` tactic.
+-/
```
Comment (semorrison) [inline Mathlib/Tactic/Basic.lean:265]:
```suggestion
/--
Tries to solve the goal using a canonical proof of `true` or the `reflexivity` tactic.
Unlike `trivial` or `trivial'`, does not use the `contradiction` tactic.
-/
```

Let's try to preserve existing documentation: tactic documentation is all being ported by hand.

File: Mathlib/Tactic/Basic.lean [hunk 3]
```diff
@@ -260,6 +260,12 @@ elab "fapply " e:term : tactic =>
 elab "eapply " e:term : tactic =>
   evalApplyLikeTactic (Meta.apply (cfg := {newGoals := ApplyNewGoals.nonDependentOnly})) e
 
+/--
+Tries to solve the goal using a canonical proof of `true` or the `reflexivity` tactic.
```
Comment (semorrison) [inline Mathlib/Tactic/Basic.lean:264]:
```suggestion
Tries to solve the goal using a canonical proof of `True`, or the `reflexivity` tactic.
```

File: test/triv.lean [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import Mathlib.Tactic.Basic
+
+example : True := by
+  triv
+
+example : 2 + 2 = 4 := by
+  triv
```
Comment (semorrison) [inline test/triv.lean:7]:
```suggestion
example : 2 + 2 = 4 := by
  triv
  
-- Verify the difference in behaviour between `triv` and `trivial`.
example (P : Prop) (h1 : P) (h2 : ¬ P) : False := by
  fail_if_success triv -- fails
  trivial -- succeeds  
```

**Comment by digama0** [comment]
@semorrison That's my fault, I told Kevin that it was fine to make PRs from a fork like in other github projects. It's not entirely clear to me whether we need to also be following the mathlib contribution strategy with the new CI setup.

**Comment by semorrison** [comment]
> @semorrison That's my fault, I told Kevin that it was fine to make PRs from a fork like in other github projects. It's not entirely clear to me whether we need to also be following the mathlib contribution strategy with the new CI setup.

No problem. It was just a few extra steps for me to merge master into this to resolve a conflict.

**Comment by kbuzzard** [comment]
We still have a conflict :-(

**Pull request closed without merging**